File: 125-Valid-Palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int r = 0, l = s.size() - 1;

        while (r < l) {
            if (!isalnum(s[r])) {
                r++;
                continue;
            }

            if (!isalnum(s[l])) {
                l--;
                continue;
            }

            if ((char)tolower(s[r]) != (char)tolower(s[l]))
                return false;

            r++, l--;
        }

        return true;
    }
};

File: 1-ArraysHashing/125-Valid-Palindrome.ts
function isAlnum(c: string): boolean {
  return !!c.match(/[a-zA-Z0-9]/);
}

function isPalindrome(s: string): boolean {
  let l = 0, r = s.length - 1

  while (l < r) {
    if (!isAlnum(s[l])) {
      l++
    } else if (!isAlnum(s[r])) {
      r--
    } else {
      if (s[l].toLowerCase() !== s[r].toLowerCase()) return false
      l++
      r--
    }
  }

  return true
};